doc/languages-frameworks/agda.section.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Agda
+author: Alex Rice (alexarice)
+date: 2020-01-06
+---
+# Agda
+
+## How to use Agda
+
+Agda can be installed with `agda.Agda`. For example:
+```
+$ nix-env -iA agda.Agda
+```
+
+To use agda with libraries, the `withPackages` function can be used. This function either takes:
++ A list of packages,
++ or a function which returns a list of packages when given the `agda` attribute set.
+
+For example, suppose we wanted a version of agda which has access to the standard library. This can be obtained with the expressions:
+
+```
+agda.withPackages [ agda.standard-library ]
+```
+
+or
+
+```
+agda.withPackages (p: [ p.standard-library ])
+```
+
+If you want to use a library in your home directory (for instance if it doesn't have a nix derivation or is a development version) then an attribute set can be passed into `agda.withPackages` with the following arguments:
+
+```
+agda.withPackages {
+  pkgs = [ agda.standard-library ];
+  homeLibraries = "path/to/my/package.agda-lib";
```

Comment:
Maybe this should also clarify whether these paths are relative or absolute. (Or whether it doesn't matter.) Maybe a link to https://agda.readthedocs.io/en/latest/tools/package-system.html#example-using-the-standard-library would be good.

Reply:
I've also been messing round and I can reproduce the error without the homeLibraries bit now. It all goes away if you include `--local-interfaces` but this shouldn't be necessary

Reply:
Hmm, I can confirm that. This means that for some reason agda is trying to put the files somewhere where they don't belong? Isn't `--local-interfaces` the default?

Also, does this give any insight:
```
$ agda -l standard-library -l mylib -i . TestUserLibrary.agda -v 10000
Importing the primitive modules.
Library dir = "/nix/store/7hbp89vwrih5hhdja12warhh5s8svn4n-Agda-2.6.1-data/share/ghc-8.8.3/x86_64-linux-ghc-8.8.3/Agda-2.6.1/lib"
SetRange is setting the current range to /nix/store/7hbp89vwrih5hhdja12warhh5s8svn4n-Agda-2.6.1-data/share/ghc-8.8.3/x86_64-linux-ghc-8.8.3/Agda-2.6.1/lib/prim/Agda/Primitive.agda:5,8-22 which is outside of the current file Nothing
/nix/store/_build: createDirectory: permission denied (Read-only
file system)
```

Reply:
Sorry I was a bit busy. I've tested this in a nix-shell with a default agda and I get the same error. Somewhere in the 2.6.0.1 to 2.6.1 change there was a change from putting interface files next to the original and putting them all in a _build directory. The build directory should go at the base of the current library you are working and next to the file otherwise, however it seems on nix, agda tries to put _build in /nix/store for some unknown reason (this only happens when you are not working in a library, which I think is why I hadn't noticed it.

I'll make an issue on agda for this and suggest we put on --local-interfaces for now on these tests?

Reply:
That sounds like a good resolution! I'll add that option to the test and clean it up.

Reply:
Unfortunately it doesn't work:
```
machine: output: Checking TestUserLibrary (/tmp/TestUserLibrary.agda).
 Checking TestModule (/tmp/mylib/src/TestModule.agda).
  Checking IO (/nix/store/r7dm2z5f4cbfpmbiiqjsi845iq91m9hx-standard-library-1.3/src/IO.agda).
   Checking Codata.Musical.Notation (/nix/store/r7dm2z5f4cbfpmbiiqjsi845iq91m9hx-standard-library-1.3/src/Codata/Musical/Notation.agda).
Failed to write interface /nix/store/r7dm2z5f4cbfpmbiiqjsi845iq91m9hx-standard-library-1.3/src/Codata/Musical/Notation.agdai.
/nix/store/r7dm2z5f4cbfpmbiiqjsi845iq91m9hx-standard-library-1.3/src/IO.agda:11,1-36
/nix/store/r7dm2z5f4cbfpmbiiqjsi845iq91m9hx-standard-library-1.3/src/Codata/Musical/Notation.agdai:
openBinaryFile: permission denied (Read-only file system)

error: command `agda -l standard-library -l mylib -i . TestUserLibrary.agda --local-interfaces` failed (exit code 1)
```
It tries to write interface files for the standard library as well now.

Reply:
Setting `--compile-dir=/tmp/` doesn't have any effect.

Reply:
This is only for compiling, not for interface files

Reply:
Isn't one of the points of this PR that `nix` will precompile the interface files of, say, the standard library? So it doesn't need to be recompiled?

Reply:
Yes, tbh I'm not sure how to fix this. We could set --local-interfaces everywhere (i.e even when building libraries)

Reply:
> Yes, tbh I'm not sure how to fix this. We could set --local-interfaces everywhere (i.e even when building libraries)

I think that's a fair workaround. After all, this `_build` directory is a fairly new feature, so it's ok to not immediately support it.